### ISSUE TITLE
Failed to fetch due to Has sum mismatch

### ISSUE THREAD

**Issue by f3wenbo** [microsoft/vscode#86858]
When I was trying to update, I got the Hash sum mismatch error on terminal.

- VSCode Version:1.40.2
- OS Version:Ubuntu LTS 18.04 (Linux x64 5.0.0-37-generic)

Steps to Reproduce:

1.sudo apt update at terminal
2.Get the following error log

```
E: Failed to fetch http://packages.microsoft.com/repos/vscode/dists/stable/main/binary-amd64/Packages.bz2  File has unexpected size (156056 != 156065). Mirror sync in progress? [IP: 10.77.8.70 8080]
   Hashes of expected file:
    - Filesize:156065 [weak]
    - SHA512:12e6dffcd247d1ff067acfa61a3c4eb6078c326c2858feb4f6f141a1d809e22292a78282414a5c103038ea80bb63cd734c7516782c9809c341335083d4e7611b
    - SHA256:c850f35f1979831ac75641978c397e51519fa22313598ec12bde5f52277044c0
    - SHA1:1f65f4a7388c22bcf7b7f30fb7bc337c60d94b2e [weak]
    - MD5Sum:4153a12e4a5629bfb14ed98f4c8ac234 [weak]
   Release file created at: Thu, 12 Dec 2019 19:59:19 +0000
E: Some index files failed to download. They have been ignored, or old ones used instead.
```

Previously:

#70584
#66339
#66238
#41117

**Comment by dpkristensen** [comment]
Also seeing this, same version of VSCode and Ubuntu as OP.

**Comment by sanjaykarki** [comment]
Same here
`Err:6 http://packages.microsoft.com/repos/vscode stable/main amd64 Packages
  File has unexpected size (156056 != 156065). Mirror sync in progress? [IP: 13.80.10.205 80]
  Hashes of expected file:
   - Filesize:156065 [weak]
   - SHA512:12e6dffcd247d1ff067acfa61a3c4eb6078c326c2858feb4f6f141a1d809e22292a78282414a5c103038ea80bb63cd734c7516782c9809c341335083d4e7611b
   - SHA256:c850f35f1979831ac75641978c397e51519fa22313598ec12bde5f52277044c0
   - SHA1:1f65f4a7388c22bcf7b7f30fb7bc337c60d94b2e [weak]
   - MD5Sum:4153a12e4a5629bfb14ed98f4c8ac234 [weak]
  Release file created at: Thu, 12 Dec 2019 19:59:19 +0000
Get:7 http://us.archive.ubuntu.com/ubuntu eoan-updates InRelease [97.5 kB]
Get:8 http://us.archive.ubuntu.com/ubuntu eoan-backports InRelease [88.8 kB]
Fetched 284 kB in 1s (215 kB/s)    
Reading package lists... Done
E: Failed to fetch http://packages.microsoft.com/repos/vscode/dists/stable/main/binary-amd64/Packages.bz2  File has unexpected size (156056 != 156065). Mirror sync in progress? [IP: 13.80.10.205 80]
   Hashes of expected file:
    - Filesize:156065 [weak]
    - SHA512:12e6dffcd247d1ff067acfa61a3c4eb6078c326c2858feb4f6f141a1d809e22292a78282414a5c103038ea80bb63cd734c7516782c9809c341335083d4e7611b
    - SHA256:c850f35f1979831ac75641978c397e51519fa22313598ec12bde5f52277044c0
    - SHA1:1f65f4a7388c22bcf7b7f30fb7bc337c60d94b2e [weak]
    - MD5Sum:4153a12e4a5629bfb14ed98f4c8ac234 [weak]
   Release file created at: Thu, 12 Dec 2019 19:59:19 +0000
E: Some index files failed to download. They have been ignored, or old ones used instead.
`

**Comment by sanjaykarki** [comment]
Same here
`
E: Failed to fetch http://packages.microsoft.com/repos/vscode/dists/stable/main/binary-amd64/Packages.bz2  File has unexpected size (156056 != 156065). Mirror sync in progress? [IP: 13.80.10.205 80]
   Hashes of expected file:
    - Filesize:156065 [weak]
    - SHA512:12e6dffcd247d1ff067acfa61a3c4eb6078c326c2858feb4f6f141a1d809e22292a78282414a5c103038ea80bb63cd734c7516782c9809c341335083d4e7611b
    - SHA256:c850f35f1979831ac75641978c397e51519fa22313598ec12bde5f52277044c0
    - SHA1:1f65f4a7388c22bcf7b7f30fb7bc337c60d94b2e [weak]
    - MD5Sum:4153a12e4a5629bfb14ed98f4c8ac234 [weak]
   Release file created at: Thu, 12 Dec 2019 19:59:19 +0000
E: Some index files failed to download. They have been ignored, or old ones used instead.

`

**Comment by moharsen** [comment]
Ditto. Ubuntu 18.04.

>E: Failed to fetch http://packages.microsoft.com/repos/vscode/dists/stable/main/binary-amd64/Packages.bz2  File has unexpected size (156056 != 156065). Mirror sync in progress? [IP: 40.76.35.62 80]
   Hashes of expected file:
    - Filesize:156065 [weak]
    - SHA512:12e6dffcd247d1ff067acfa61a3c4eb6078c326c2858feb4f6f141a1d809e22292a78282414a5c103038ea80bb63cd734c7516782c9809c341335083d4e7611b
    - SHA256:c850f35f1979831ac75641978c397e51519fa22313598ec12bde5f52277044c0
    - SHA1:1f65f4a7388c22bcf7b7f30fb7bc337c60d94b2e [weak]
    - MD5Sum:4153a12e4a5629bfb14ed98f4c8ac234 [weak]
   Release file created at: Thu, 12 Dec 2019 19:59:19 +0000
E: Some index files failed to download. They have been ignored, or old ones used instead.

**Comment by xgdgsc** [comment]
Same with Ubuntu 16.04

**Comment by scallensc** [comment]
Same 19.10

**Comment by pffang** [comment]
Same with Ubuntu 16.04

**Comment by taoqf** [comment]
Debian 10

**Comment by qingwang237** [comment]
mint 19.2

**Comment by JFitzy1321** [comment]
same on Pop 19.10

**Comment by mrizkimaulidan** [comment]
```
E: Failed to fetch https://packages.microsoft.com/repos/vscode/dists/stable/main/binary-amd64/Packages.bz2  File has unexpected size (156056 != 156065). Mirror sync in progress? [IP: 13.76.190.189 443]
   Hashes of expected file:
    - Filesize:156065 [weak]
    - SHA512:12e6dffcd247d1ff067acfa61a3c4eb6078c326c2858feb4f6f141a1d809e22292a78282414a5c103038ea80bb63cd734c7516782c9809c341335083d4e7611b
    - SHA256:c850f35f1979831ac75641978c397e51519fa22313598ec12bde5f52277044c0
    - SHA1:1f65f4a7388c22bcf7b7f30fb7bc337c60d94b2e [weak]
    - MD5Sum:4153a12e4a5629bfb14ed98f4c8ac234 [weak]
   Release file created at: Thu, 12 Dec 2019 19:59:19 +0000
E: Some index files failed to download. They have been ignored, or old ones used instead.
```

Same problem, Kali Linux Light

**Comment by gustavocipriani** [comment]
Same here. Ubuntu 18.04.3.

> Reading package lists... Done 
> N: Skipping acquire of configured file 'contrib/binary-i386/Packages' as repository 'https://download.virtualbox.org/virtualbox/debian bionic InRelease' doesn't support architecture 'i386'
> E: Failed to fetch https://packages.microsoft.com/repos/vscode/dists/stable/main/binary-amd64/Packages.bz2  File has unexpected size (156056 != 156065). Mirror sync in progress? [IP: 40.76.35.62 443]
>    Hashes of expected file:
>     - Filesize:156065 [weak]
>     - SHA512:12e6dffcd247d1ff067acfa61a3c4eb6078c326c2858feb4f6f141a1d809e22292a78282414a5c103038ea80bb63cd734c7516782c9809c341335083d4e7611b
>     - SHA256:c850f35f1979831ac75641978c397e51519fa22313598ec12bde5f52277044c0
>     - SHA1:1f65f4a7388c22bcf7b7f30fb7bc337c60d94b2e [weak]
>     - MD5Sum:4153a12e4a5629bfb14ed98f4c8ac234 [weak]
>    Release file created at: Thu, 12 Dec 2019 19:59:19 +0000
> E: Some index files failed to download. They have been ignored, or old ones used instead.

**Comment by mag911** [comment]
Same issue here with Debian Buster

**Comment by benizra-yahoo** [comment]
Same error with Linux Mint 19.1 Tessa

**Comment by IvanRoman** [comment]
Reporting in with the same issue. Ubuntu 18.04.3

**Comment by soominjung** [comment]
Similar error with 2 Ubuntu machines

- VSCode Version: 1.40.1
- OS Version:Ubuntu LTS 16.04

- VSCode Version:1.41.1
- OS Version:Ubuntu LTS 16.04

> E: Failed to fetch http://packages.microsoft.com/repos/vscode/dists/stable/main/binary-amd64/Packages.bz2  Hash Sum mismatch
> E: Some index files failed to download. They have been ignored, or old ones used instead.

**Comment by yankyhoffman** [comment]
Ubuntu 19.10 here, however, I updated successfully within 30 minutes of the 1.41 release on my work PC which is also Ubuntu 19.10. getting the error now on my laptop.

I am getting the issue for the dotnet core apt repo as well, so it might be microsofts servers...

**Comment by imreverse** [comment]
same on KDE Neon 5.17

**Comment by RMacfarlane** [comment]
I've contacted the linux repo team about this and am waiting for them to resolve it.

**Comment by furai** [comment]
https://packages.microsoft.com/repos/vscode/dists/stable/InRelease
https://packages.microsoft.com/repos/vscode/dists/stable/Release
In these files values changed for 
```
Origin: vscode stable
Label: vscode stable
Suite: stable
Codename: stable
``` 
That's probably why there are hash sum mismatches.

**Comment by tedlz** [comment]
same on Debian 10

**Comment by peter279k** [comment]
I've the same issue on Ubuntu 18.04.3 and 16.04.6 LTS.

**Comment by istinj** [comment]
Same problem on Ubuntu Mate 18.04.03

**Comment by pieczarkazsolt** [comment]
same here (ubuntu 19.10)

` https://packages.microsoft.com/repos/vscode/dists/stable/main/binary-amd64/Packages.bz2  File has unexpected size (156354 != 156065). Mirror sync in progress? [IP: 13.74.252.37 443]
   Hashes of expected file:
    - Filesize:156065 [weak]
    - SHA512:12e6dffcd247d1ff067acfa61a3c4eb6078c326c2858feb4f6f141a1d809e22292a78282414a5c103038ea80bb63cd734c7516782c9809c341335083d4e7611b
    - SHA256:c850f35f1979831ac75641978c397e51519fa22313598ec12bde5f52277044c0
    - SHA1:1f65f4a7388c22bcf7b7f30fb7bc337c60d94b2e [weak]
    - MD5Sum:4153a12e4a5629bfb14ed98f4c8ac234 [weak]
   Release file created at: Thu, 12 Dec 2019 19:59:19 +0000
`

**Comment by MucTepKOT** [comment]
**same problem (linux mint lastest)**

E: Failed to fetch https://packages.microsoft.com/repos/vscode/dists/stable/main/binary-amd64/Packages.bz2  File has unexpected size (156354 != 156065). Mirror sync in progress? [IP: 13.74.252.37 443]
   Hashes of expected file:
    - Filesize:156065 [weak]
    - SHA512:12e6dffcd247d1ff067acfa61a3c4eb6078c326c2858feb4f6f141a1d809e22292a78282414a5c103038ea80bb63cd734c7516782c9809c341335083d4e7611b
    - SHA256:c850f35f1979831ac75641978c397e51519fa22313598ec12bde5f52277044c0
    - SHA1:1f65f4a7388c22bcf7b7f30fb7bc337c60d94b2e [weak]
    - MD5Sum:4153a12e4a5629bfb14ed98f4c8ac234 [weak]
   Release file created at: Thu, 12 Dec 2019 19:59:19 +0000
E: Some index files failed to download. They have been ignored, or old ones used instead.

**Comment by pieczarkazsolt** [comment]
https://en.wikipedia.org/wiki/Friday_the_13th :D

**Comment by yucer** [comment]
I receive the exact error in *Linux Mint 19.2 Tina*

**Comment by marcmorente** [comment]
Same here on Debian 10 Buster

**Comment by AhmadRoshanfar** [comment]
$ software-properties-gtk
-> **other software** window 
-> remove Microsoft link
$ sudo apt-get update

**Comment by ksze** [comment]
> https://packages.microsoft.com/repos/vscode/dists/stable/InRelease
> https://packages.microsoft.com/repos/vscode/dists/stable/Release
> In these files values changed for
> 
> ```
> Origin: vscode stable
> Label: vscode stable
> Suite: stable
> Codename: stable
> ```
> 
> That's probably why there are hash sum mismatches.

Yeah, I don't understand what this is all about either.

When I run `sudo apt update`, I get these messages:
```
E: Repository 'https://packages.microsoft.com/ubuntu/19.04/prod disco InRelease' changed its 'Origin' value from 'microsoft-ubuntu-disco-prod disco' to 'vscode stable'
E: Repository 'https://packages.microsoft.com/ubuntu/19.04/prod disco InRelease' changed its 'Label' value from 'microsoft-ubuntu-disco-prod disco' to 'vscode stable'
E: Repository 'https://packages.microsoft.com/ubuntu/19.04/prod disco InRelease' changed its 'Suite' value from 'disco' to 'stable'
E: Repository 'https://packages.microsoft.com/ubuntu/19.04/prod disco InRelease' changed its 'Codename' value from 'disco' to 'stable'
N: This must be accepted explicitly before updates for this repository can be applied. See apt-secure(8) manpage for details.
Do you want to accept these changes and continue updating from this repository? [y/N]
```
I answered `y` to the last question, but it doesn't seem to do anything, I still get this message some lines later:
```
W: Conflicting distribution: https://packages.microsoft.com/ubuntu/19.04/prod disco InRelease (expected disco but got stable)
```

**Comment by Okroshiashvili** [comment]
Same problem with Linux Mint 19.

**Comment by drihu** [comment]
Same here on Ubuntu 18.04 Bionic

**Comment by yakubenko** [comment]
Same here :( Ubuntu 18.04

**Comment by MohiburShuvo** [comment]
File has unexpected size (156354 != 156065). Mirror sync in progress? [IP: 13.75.127.55 80]
  Hashes of expected file:
   - Filesize:156065 [weak]
   - SHA512:12e6dffcd247d1ff067acfa61a3c4eb6078c326c2858feb4f6f141a1d809e22292a78282414a5c103038ea80bb63cd734c7516782c9809c341335083d4e7611b
   - SHA256:c850f35f1979831ac75641978c397e51519fa22313598ec12bde5f52277044c0
   - SHA1:1f65f4a7388c22bcf7b7f30fb7bc337c60d94b2e [weak]
   - MD5Sum:4153a12e4a5629bfb14ed98f4c8ac234 [weak]
  Release file created at: Thu, 12 Dec 2019 19:59:19 +0000
Hit:16 http://packages.linuxmint.com tina Release
Fetched 252 kB in 26s (9,692 B/s)
Reading package lists... Done
E: Failed to fetch http://packages.microsoft.com/repos/vscode/dists/stable/main/binary-amd64/Packages.bz2  File has unexpected size (156354 != 156065). Mirror sync in progress? [IP: 13.75.127.55 80]
   Hashes of expected file:
    - Filesize:156065 [weak]
    - SHA512:12e6dffcd247d1ff067acfa61a3c4eb6078c326c2858feb4f6f141a1d809e22292a78282414a5c103038ea80bb63cd734c7516782c9809c341335083d4e7611b
    - SHA256:c850f35f1979831ac75641978c397e51519fa22313598ec12bde5f52277044c0
    - SHA1:1f65f4a7388c22bcf7b7f30fb7bc337c60d94b2e [weak]
    - MD5Sum:4153a12e4a5629bfb14ed98f4c8ac234 [weak]
   Release file created at: Thu, 12 Dec 2019 19:59:19 +0000
E: Some index files failed to download. They have been ignored, or old ones used instead. 

Same with Linux Mint 19.2 Tina

**Comment by ajithcodesit** [comment]
Same error occurring during update on Ubuntu 18.04.

`Err:11 http://packages.microsoft.com/repos/vscode stable/main amd64 Packages   
  File has unexpected size (156354 != 156065). Mirror sync in progress? [IP: 13.76.190.189 80]
  Hashes of expected file:
   - Filesize:156065 [weak]
   - SHA512:12e6dffcd247d1ff067acfa61a3c4eb6078c326c2858feb4f6f141a1d809e22292a78282414a5c103038ea80bb63cd734c7516782c9809c341335083d4e7611b
   - SHA256:c850f35f1979831ac75641978c397e51519fa22313598ec12bde5f52277044c0
   - SHA1:1f65f4a7388c22bcf7b7f30fb7bc337c60d94b2e [weak]
   - MD5Sum:4153a12e4a5629bfb14ed98f4c8ac234 [weak]
  Release file created at: Thu, 12 Dec 2019 19:59:19 +0000
`

**Comment by rushxs7** [comment]
same issue on ubuntu 19.10, tried manually installing the .deb but no help

**Comment by VilhelmPrytz** [comment]
Also have the same issue as everybody else, across three different PC's.

**Comment by koyadovic** [comment]
Same in Debian 10

Err:12 http://packages.microsoft.com/repos/vscode stable/main amd64 Packages
  File has unexpected size (156354 != 156065). Mirror sync in progress? [IP: 13.80.10.205 80]
  Hashes of expected file:
   - Filesize:156065 [weak]
   - SHA512:12e6dffcd247d1ff067acfa61a3c4eb6078c326c2858feb4f6f141a1d809e22292a78282414a5c103038ea80bb63cd734c7516782c9809c341335083d4e7611b
   - SHA256:c850f35f1979831ac75641978c397e51519fa22313598ec12bde5f52277044c0
   - SHA1:1f65f4a7388c22bcf7b7f30fb7bc337c60d94b2e [weak]
   - MD5Sum:4153a12e4a5629bfb14ed98f4c8ac234 [weak]
  Release file created at: Thu, 12 Dec 2019 19:59:19 +0000

**Comment by gjsjohnmurray** [comment]
> 
Earlier from @RMacfarlane:
 
> I've contacted the linux repo team about this and am waiting for them to resolve it.

Pinging @sbatten as endgame master for this release. This seems to be taking a long time to resolve, and is resulting in a lot of dup issues. The bot isn't doing a great job of spotting them either.

**Comment by mshzsh** [comment]
Same problem on Ubuntu 18.04.3 LTS

**Comment by medmedin2014** [comment]
Same problem on Kubuntu 19.10 - 64bits

**Comment by tnatanael** [comment]
+1 Lubuntu 18.04

**Comment by angelosds** [comment]
Same problem on Elementary OS 5.1

**Comment by Sander-Verk** [comment]
+1 On ubuntu 16.04

**Comment by grillorafael** [comment]
Also having the same issues installing msodbcsql17 on debian buster

**Comment by mantans** [comment]
Same here on Debian 10 Buster

**Comment by andrenfarias** [comment]
![vscode-bug-sugestao](https://user-images.githubusercontent.com/5272321/70802855-5f075180-1d91-11ea-9c89-c17396041c78.png)

**Comment by SouBanerjee** [comment]
E: Failed to fetch http://packages.microsoft.com/repos/vscode/dists/stable/main/binary-amd64/Packages.bz2  File has unexpected size (156354 != 156065). Mirror sync in progress? [IP: 13.76.190.189 80]
   Hashes of expected file:
    - Filesize:156065 [weak]
    - SHA512:12e6dffcd247d1ff067acfa61a3c4eb6078c326c2858feb4f6f141a1d809e22292a78282414a5c103038ea80bb63cd734c7516782c9809c341335083d4e7611b
    - SHA256:c850f35f1979831ac75641978c397e51519fa22313598ec12bde5f52277044c0
    - SHA1:1f65f4a7388c22bcf7b7f30fb7bc337c60d94b2e [weak]
    - MD5Sum:4153a12e4a5629bfb14ed98f4c8ac234 [weak]
   Release file created at: Thu, 12 Dec 2019 19:59:19 +0000
E: Some index files failed to download. They have been ignored, or old ones used instead.


Ubuntu 18.04

**Comment by jaspreet57** [comment]
It is everywhere now :).... I have removed the vscode sources for now until this is fixed.
`sudo rm /etc/apt/sources.list.d/vscode.list`

**Comment by tommituura** [comment]
Same problem here, xUbuntu 18.04.3

**Comment by iricigor** [comment]
This is happening every month. 

#74826
    #70584
    #66339
    #66238
    #41117
    #25883

**Comment by iricigor** [comment]
Here also...

**Comment by habebit** [comment]
Same problem on Debian 10 Buster.

**Comment by Senshi111** [comment]
Same Debian Buster

**Comment by rpellerin** [comment]
Ok maybe people should stop posting messages such as "Same on <platform>". I guess at this point they know most (if not all) Debian-based platforms are affected.

**Comment by jbruni** [comment]
Please resolve this issue, Microsoft guys. Thank you.

**Comment by ankush981** [comment]
@rpellerin Maybe not. It does increase the noise, I agree, but it's a great metric of how pervasive the problem is. Unless we have a voting/input system of sorts ("Yes, I have this problem, too!"), the number and intensity of comments is the only way :-)

**Comment by ankush981** [comment]
And oh, I'm too having this problem with Xubuntu 18.04. :laughing:

**Comment by rpellerin** [comment]
> @rpellerin Maybe not. It does increase the noise, I agree, but it's a great metric of how pervasive the problem is. Unless we have a voting/input system of sorts ("Yes, I have this problem, too!"), the number and intensity of comments is the only way :-)

At least people should look for duplicates and thumb up messages mentioning their own distro instead of re-posting.

**Comment by santiago-ld** [comment]
Same here, Ubuntu 19.10

**Comment by gregorthebigmac** [comment]
I love all the people chiming in with "I'm having the same issue on [distro] v[XX.xx] [desktop environment] as well!" Yes, you're _all_ going to have the issue. The vscode team screwed up their PGP key. It's not going to work for _any_ of us until they fix the key. It will probably be resolved by tomorrow, because it's not that difficult of a fix. So just manually ignore the vscode update until it's fixed. If you don't know how to do that, read [this](https://tecadmin.net/exclude-packages-from-apt-upgrade/)

**Comment by BrainStone** [comment]
dotnet core is affected too by this fuck up: dotnet/core#3988

**Comment by abhishekjnvk** [comment]
It is everywhere now :).... I have removed the vscode sources for now until this is fixed.
sudo rm /etc/apt/sources.list.d/vscode.list

**Comment by ksze** [comment]
> I love all the people chiming in with "I'm having the same issue on [distro] v[XX.xx] [desktop environment] as well!" Yes, you're _all_ going to have the issue. The vscode team screwed up their PGP key. It's not going to work for _any_ of us until they fix the key. It will probably be resolved by tomorrow, because it's not that difficult of a fix. So just manually ignore the vscode update until it's fixed. If you don't know how to do that, read [this](https://tecadmin.net/exclude-packages-from-apt-upgrade/)
> 
> Edit: Here's a more comprehensive guide to ignoring repos, if you're interested. https://www.tecmint.com/disable-lock-blacklist-package-updates-ubuntu-debian-apt/

I ... don't think this has to do with PGP key?

**Comment by pffang** [comment]
Yeah, I am absolutely clear what the problem is, but also know how to solve it.
But I still come here and write down "Same on ...". This is just a euphemism to express: "Hey, you guys screwed up again."
Yeah, again and again. 
You guys represent a great software company. Don't let me so dispoint. Just setup an automated test please.

**Comment by ksze** [comment]
@f3wenbo Can you fix the issue title? The typo doesn't help the bot (and other users) finding duplicates, I suspect.

**Comment by ashwinee2591** [comment]
> When I was trying to update, I got the Hash sum mismatch error on terminal.
> 
>     * VSCode Version:1.40.2
> 
>     * OS Version:Ubuntu LTS 18.04 (Linux x64 5.0.0-37-generic)
> 
> 
> Steps to Reproduce:
> 
> 1.sudo apt update at terminal
> 2.Get the following error log
> 
> ```
> E: Failed to fetch http://packages.microsoft.com/repos/vscode/dists/stable/main/binary-amd64/Packages.bz2  File has unexpected size (156056 != 156065). Mirror sync in progress? [IP: 10.77.8.70 8080]
>    Hashes of expected file:
>     - Filesize:156065 [weak]
>     - SHA512:12e6dffcd247d1ff067acfa61a3c4eb6078c326c2858feb4f6f141a1d809e22292a78282414a5c103038ea80bb63cd734c7516782c9809c341335083d4e7611b
>     - SHA256:c850f35f1979831ac75641978c397e51519fa22313598ec12bde5f52277044c0
>     - SHA1:1f65f4a7388c22bcf7b7f30fb7bc337c60d94b2e [weak]
>     - MD5Sum:4153a12e4a5629bfb14ed98f4c8ac234 [weak]
>    Release file created at: Thu, 12 Dec 2019 19:59:19 +0000
> E: Some index files failed to download. They have been ignored, or old ones used instead.
> ```
> 
> Previously:
> 
> #70584
> #66339
> #66238
> #41117

same with Linux Mint 19.2

**Comment by ashwinee2591** [comment]
Same with Linux Mint 19.2.

**Comment by yakubenko** [comment]
> Yeah, I am absolutely clear what the problem is, but also know how to solve it.
> But I still come here and write down "Same on ...". This is just a euphemism to express: "Hey, you guys screwed up again."
> Yeah, again and again.
> You guys represent a great software company. Don't let me so dispoint. Just setup an automated test please.

I've been using vscode for quite a while and this is the only issue I've experienced so far. Pls, be patient.

**Comment by emnovaes** [comment]
Debian 10.
     Puxa, esse erro já ocorreu em 2018 (by google)

Err:11 http://packages.microsoft.com/repos/vscode stable/main amd64 Packages
  File has unexpected size (156354 != 156065). Mirror sync in progress? [IP: 40.76.35.62 80]
  Hashes of expected file:
   - Filesize:156065 [weak]

**Comment by abhishekjnvk** [comment]
Simply remove removed the vscode sources for now and add when it is fixed.
sudo rm /etc/apt/sources.list.d/vscode.list

**Comment by likhvar** [comment]
Ubuntu 18.4

**Comment by emnovaes** [comment]
But removing this file (vscode.list), there are will created again?

**Comment by yankyhoffman** [comment]
For all people trying to help by saying "remove the vscode.list file" don't forget, the issue created here is a vscode (Microsoft (as it applies to dotnet-core as well)) issue, not an apt issue, and removing the .list file will only allow us to update all other packages vscode won't be updated unless the vscode.list file exists and works. (Yes I know we can install the update using the deb, but that is not the issue here either)

**Comment by martindholmes** [comment]
Ditto Ubuntu 19.10.

**Comment by mainframed** [comment]
Same on Xubuntu Ubuntu 19.10

**Comment by pffang** [comment]
> 
> 
> > Yeah, I am absolutely clear what the problem is, but also know how to solve it.
> > But I still come here and write down "Same on ...". This is just a euphemism to express: "Hey, you guys screwed up again."
> > Yeah, again and again.
> > You guys represent a great software company. Don't let me so dispoint. Just setup an automated test please.
> 
> I've been using vscode for quite a while and this is the only issue I've experienced so far. Pls, be patient.

Just google "vscode Has sum mismatch". You will know what I talk about.
I'm a programer. I understand and accept it has some bugs.
But, this is a small issue and easy to be discovered by automated test. I believe it's not difficult for them technically and resourceally.
They promised to improve it. But sadly, they didn't.

**Comment by Hellkyte** [comment]
Same 16.04 ... our ci/cd is failing trying to install the odbc drivers for sqlserver....

**Comment by franksmcb** [comment]
Issue exists on Ubuntu MATE 19 as well.

**Comment by urielhass** [comment]
same thing here Linux Mint 19.2 Cinnamon

**Comment by an0mal1a** [comment]
on MX Linux the same. But i commented the repository url in /etc/apt/sources.list.d/vscode.list, then i could update without problem, for now.

**Comment by cbRoy** [comment]
Ubuntu 18.04 brought me here to complain

**Comment by maurom** [comment]
It seems the signed release file http://packages.microsoft.com/repos/vscode/dists/stable/InRelease (**InRelease**) is old and does not match the files on the repo.

```
URL=http://packages.microsoft.com/repos/vscode/dists/stable

curl -s $URL/InRelease | grep Packages.bz2 |head -1
 4153a12e4a5629bfb14ed98f4c8ac234   156065 main/binary-amd64/Packages.bz2   <-- wrong size
curl -s $URL/Release | grep Packages.bz2 |head -1
 de01bf2799b61b2a6eb891b6c6f173e1   156354 main/binary-amd64/Packages.bz2   <-- ok
curl -s -I $URL/main/binary-amd64/Packages.bz2 | grep Length
Content-Length: 156354   <-- real size

curl -s $URL/InRelease | grep Date
Date: Thu, 12 Dec 2019 19:59:19 UTC   <--- older
curl -s $URL/Release | grep Date
Date: Fri, 13 Dec 2019 14:15:54 UTC   <-- newer

```

**Comment by Hellkyte** [comment]
Looks like this might be fixed
```
Get:2 http://us.archive.ubuntu.com/ubuntu xenial-updates InRelease [109 kB]
Get:3 http://us.archive.ubuntu.com/ubuntu xenial-backports InRelease [107 kB]
Get:4 http://security.ubuntu.com/ubuntu xenial-security InRelease [109 kB]
Get:5 https://packages.microsoft.com/ubuntu/16.04/prod xenial InRelease [4,003 B]
Get:6 https://packages.microsoft.com/ubuntu/16.04/prod xenial/main amd64 Packages [110 kB]
Hit:7 https://packages.gitlab.com/runner/gitlab-runner/ubuntu xenial InRelease
Fetched 440 kB in 1s (232 kB/s)
Reading package lists... Done
```

**Comment by lodyman93** [comment]
> Looks like this might be fixed
> 
> ```
> Get:2 http://us.archive.ubuntu.com/ubuntu xenial-updates InRelease [109 kB]
> Get:3 http://us.archive.ubuntu.com/ubuntu xenial-backports InRelease [107 kB]
> Get:4 http://security.ubuntu.com/ubuntu xenial-security InRelease [109 kB]
> Get:5 https://packages.microsoft.com/ubuntu/16.04/prod xenial InRelease [4,003 B]
> Get:6 https://packages.microsoft.com/ubuntu/16.04/prod xenial/main amd64 Packages [110 kB]
> Hit:7 https://packages.gitlab.com/runner/gitlab-runner/ubuntu xenial InRelease
> Fetched 440 kB in 1s (232 kB/s)
> Reading package lists... Done
> ```

no it's not fixed yet

**Comment by aabouissa** [comment]
Same problem on Debian 10, but i removed currently the source list using:

`sudo rm /etc/apt/sources.list.d/vscode.list`

I'm waiting to fix this error.

**Comment by cbRoy** [comment]
confirmed. not fixed.

**Comment by jasjuang** [comment]
Same problem here on Ubuntu 18.04.

**Comment by 333fps** [comment]
Fixed :)

**Comment by jasjuang** [comment]
For me, now `sudo apt update` shows 

```
N: Skipping acquire of configured file 'main/binary-i386/Packages' as repository 'https://packages.microsoft.com/ubuntu/18.04/prod bionic InRelease' doesn't support architecture 'i386'
```

**Comment by hiagodotme** [comment]
Was with the same problem, apparently normalized.

**Comment by tommituura** [comment]
Working nicely now, xUbuntu 18.04.3. No changes to any apt sources at any point.

**Comment by vazgen6** [comment]
Finally fixed, running GNU/Linux Mint (based on Ubuntu Bionic)

**Comment by RMacfarlane** [comment]
This should now be fixed. Thanks everyone for your patience.

**Comment by gregorthebigmac** [comment]
> I ... don't think this has to do with PGP key?

After re-reading my comment, I think it's a GPG key, but otherwise, yes. Are you sure you read the error output after running "sudo apt-get update"? Correct me if I'm wrong here, but that's what happens when you fuck up your PGP/GPG key and upload it to the apt repo. The hash key is fucked, and apt no longer trusts it because your key doesn't match the one in the repo. Now, you can choose to trust it anyway with "sudo apt-get --allow-unauthenticated upgrade" as per [this](https://askubuntu.com/questions/74345/how-do-i-bypass-ignore-the-gpg-signature-checks-of-apt) explanation, but I wouldn't recommend that, because you can't guarantee it's coming from a trusted source (in this case, MS). This is why I instead told people to hold off on updating vscode until MS fixes it, and they already have.

**Comment by emnovaes** [comment]
resolved

**Comment by deyvisonrocha** [comment]
Snap install without problems.

**Comment by drihu** [comment]
Fixed on Ubuntu 18.04 Bionic :smile:

**Comment by Tyriar** [comment]
FYI for a little more transparency, this happened because changes were pushed to the system that drives https://packages.microsoft.com/ (for the majority of MS Linux offerings) and they caused issues with metadata sync. We have similar issues periodically where some repo files don't regenerate without their team doing some manual intervention. Unfortunately all we (the VS Code team) can do is ping them when things go wrong and hope that the underlying problems get fixed eventually.
